bork: init at 0.4.0-unstable-2025-04-18

jonhermansen opened the pull request:
https://github.com/kristoff-it/bork

Bork is a TUI chat client tailored for livecoding on Twitch.

My first Zig package. Used zon2nix to pull dependencies into the sandbox.

There hasn't been an official release in a while, so I'm pulling the latest commit from master branch.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jonhermansen committed:
bork: init at 0.4.0-unstable-2025-04-18

yzhou216 commented on pkgs/by-name/bo/bork/package.nix:
```diff
@@ -0,0 +1,42 @@
+let
+  zig = zig_0_14;
+in
+stdenv.mkDerivation rec {
```
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

jonhermansen commented on pkgs/by-name/bo/bork/package.nix:
```diff
@@ -0,0 +1,42 @@
+    ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
+  '';
+
+  meta = with lib; {
+    description = "A TUI chat client tailored for livecoding on Twitch.";
+    homepage = "https://github.com/kristoff-it/bork";
+    changelog = "https://github.com/kristoff-it/bork/compare/0.4.0...master";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jonhermansen ];
+    platforms = platforms.unix;
+    mainProgram = "bork";
+  };
```
I'm in touch with the author and I will try to ask for a new version, so the package can use a git tag in the future.

jonhermansen replied:
I decided to make the changelog link to the GitHub releases URL (without any version interpolated), so finalAttrs is not needed, so I removed it.

acid-bong commented on pkgs/by-name/bo/bork/package.nix:
```diff
@@ -0,0 +1,42 @@
+let
+  zig = zig_0_14;
+in
+stdenv.mkDerivation rec {
```
Doesn't matter, if nothing is reused, both finalAttrs and rec are useless

jonhermansen replied:
I ended up using finalAttrs because I plan to send some changes upstream, will ask for a release, and will reference finalAttrs in the changelog URL (if not also the src git tag)

jonhermansen commented:
Thanks @acid-bong and @yzhou216 , I believe I've addressed all the comments. I also amended the commit message and PR title to reflect the fixed version number.

acid-bong commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 431913`

Logs: https://github.com/acid-bong/nixpkgs-review-gha/actions/runs/16822639852


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bork</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bork</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bork</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bork</li>
  </ul>
</details>

acid-bong reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.
